Lib/test/test_sys.py

```diff
@@ -71,6 +71,26 @@ def baddisplayhook(obj):
             code = compile("42", "<string>", "single")
             self.assertRaises(ValueError, eval, code)
 
+class ExcInfoTest(unittest.TestCase):
+    def test_exc_info_no_exception(self):
+        self.assertEqual(sys.exc_info(), (None, None, None))
+        self.assertEqual(sys.exception(), None)
+
+    def test_exc_info_with_exception(self):
+        def f():
+            raise ValueError(42)
```

Comment:
Can you please add a test on ``sys.exception()`` where an exception **class** is raised, rather than an instance: ``raise ValueError``? If possible, only call ``sys.exception()`` in the exept block, not ``sys.exc_info()``, to make sure that ``sys.exception()`` alone is about the create the instance.

Reply:
Good point, I added the tests. 

Currently these function just incref and return the value (the instance is created when the exception is caught, in ceval.c just after the exception_unwind label). But that's an implementation detail).